src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -457,7 +501,7 @@ class DemuxFastqs
 
 /** A writer than writes [[DemuxRecord]]s */
 private trait DemuxWriter extends Closeable {
-  def add(rec: DemuxRecord): Unit
+  def add(rec: DemuxRecord): Any
```

Comment:
so that it returns the record output (either `SamRecord` or `FastqRecord`) for testing.  I suppose in the future we can make this a type parameter.

Reply:
No reason not to do it now right?  Would take 1 minute to do.